Upgrade Go to 1.25.5 to resolve GO-2025-4155 vulnerability

### Description

**Observed Behavior**:

The CI `vulncheck` step fails with the following vulnerability:

```
Vulnerability #1: GO-2025-4155
    Excessive resource consumption when printing error string for host
    certificate validation in crypto/x509
  More info: https://pkg.go.dev/vuln/GO-2025-4155
  Standard library
    Found in: crypto/x509@go1.25.4
    Fixed in: crypto/x509@go1.25.5
    Example traces found:
      #1: pkg/controllers/nodepool/validation/controller.go:58:114: validation.Controller.Reconcile calls x509.HostnameError.Error
```

This vulnerability affects the Go standard library's `crypto/x509` package and could lead to excessive resource consumption when printing error strings for host certificate validation.

**Expected Behavior**:

The `vulncheck` step should pass without any vulnerabilities reported.

**Reproduction Steps** (Please include YAML):

1. Clone the repository
2. Run `make vulncheck`
3. Observe the GO-2025-4155 vulnerability error

**Versions**:
- Go Version: 1.25.4 (current in go.mod: `go 1.25.3`)
- Kubernetes Version: N/A (affects all versions)
- Vulnerability: GO-2025-4155
- Fix Available: Go 1.25.5

**Solution**:

Update `go.mod` to specify Go 1.25.5:
```diff
- go 1.25.3
+ go 1.25.5
```

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module sigs.k8s.io/karpenter
 
-go 1.25.3
+go 1.25.5
 
 require (
 	github.com/Pallinder/go-randomdata v1.2.0

